.github/workflows/haskell.yml

```diff
@@ -236,7 +236,7 @@ jobs:
       uses: input-output-hk/actions/haskell@latest
       with:
         ghc-version: ${{ matrix.ghc }}
-        cabal-version: latest
+        cabal-version: 3.12.1.0
```

Comment:
Could please also update in flake.nix, so we have a consistent experience locally and on hydra as well:https://github.com/IntersectMBO/cardano-ledger/blob/6c9f88f328305e16f916765d3aa0b49d2cb047bf/flake.nix#L104

Reply:
Maybe we should change both to `3.10.3.0` instead: https://github.com/IntersectMBO/cardano-ledger/commit/27a46ec6269f428ca59afa12c40f8b2f51b66bf0

Reply:
The only thing that was blocking us from upgrading to `3.12.1.0` was https://github.com/input-output-hk/actions/issues/27 which seems to be fixed no, so I don't see a reason why we should not upgrade to latest cabal version now.